### ISSUE TITLE
Crash on ARK running CM13

### ISSUE THREAD

**Issue by dimqua** [chenxiaolong/DualBootPatcher#727]
The app (latest builds, at least) is constantly crashed when I go to 'Firmwares' on my ARK Benefit A3 running CyanogenMod 13.
Logcat: 
https://privatebin.net/?fa4a1e5bce936738#6eM5NduydMmfBYEpuhyuQ4xx3jVvUK2rpRTXp1dJxE0=

And when I try to patch a zip-file with any firmware, I get crash too.

**Comment by chenxiaolong** [comment]
Thanks! I'm looking into the issue now.

**Comment by chenxiaolong** [comment]
@dimqua Could you upload the following files?

* `/data/tombstones/tombstone_08`
* `/sdcard/MultiBoot/primary/boot.img`

**Comment by dimqua** [comment]
* [/data/tombstones/tombstone_08](https://cloud.unixcorn.org/s/sNs039fjQI66zBc/download)
* Directory `/sdcard/MultiBoot/primary/` does not exist.

**Comment by chenxiaolong** [comment]
Thanks, it looks like the tombstones file doesn't contain enough information.

Can you try the debug build here and then upload `/data/tombstones` again? https://dbp.noobdev.io/DualBootPatcherAndroid-9.2.0.r339.ge167fe8f.CRASH_DEBUG.apk

I will take a look at it after work.

**Comment by dimqua** [comment]
[tombstones](https://cloud.unixcorn.org/s/ylBoyIlxT4Ln9r5)

**Comment by chenxiaolong** [comment]
Thanks, that's very helpful. I'm also able to reproduce the issue on my old Note 4 now. (The crash doesn't seem to happen on every device.)

**Comment by yshalsager** [comment]
@chenxiaolong  crashing issue is here. MI Max and nearly all devices I support

**Comment by chenxiaolong** [comment]
Found the problem. `_ZNSs4_Rep20_S_empty_rep_storageE` is being put in the BSS section of the libraries instead of being created as a unique global symbol. libstdc++ (ie. `gnustl_shared`) does this as an optimization for empty strings. However, each library (libmbcommon, libmbbootimg) has its own `_S_empty_rep_storage` symbol, which causes the global variable to be deallocated when passing an std::string between libraries.
```
[armeabi-v7a] nm libmbcommon.so | grep _ZNSs4_Rep20_S_empty_rep_storageE
001040b8 b _ZNSs4_Rep20_S_empty_rep_storageE
```

This is not an issue with the DualBootPatcher code. Something is wrong with how things are being compiled and linked. I'll see if I can find a workaround.

---
(#683 and #726 are probably related to this as well)

**Comment by wazeeahmed** [comment]
A lot of device is facing crash issue in dbp . You just didn't took a look
at those issues :(

**Comment by chenxiaolong** [comment]
@dimqua @MisiekDP Please give the test build at #738 a try and let me know if it still crashes for you.

**Comment by MisiekDP** [comment]
@chenxiaolong, this build 9.2.0.r345.gbda23dec.PR738 works for me perfectly, without crashes :)
I will test more but now I see that I don't have previous crashes :)

**Comment by chenxiaolong** [comment]
Fixed by #738.
